packages/dd-trace/test/opentracing/span.spec.js

```diff
@@ -133,7 +133,25 @@ describe('Span', () => {
       span.setBaggageItem('foo', 'bar')
 
       expect(span.context()._baggageItems).to.have.property('foo', 'bar')
-      expect(parent._baggageItems).to.have.property('foo', 'bar')
+      expect(parent._baggageItems).to.not.have.property('foo', 'bar')
+    })
+
+    it('should pass baggage items to future causal spans', () => {
+      const parent = {
+        traceId: '123',
+        spanId: '456',
+        _baggageItems: {
+          'foo': 'bar'
+        },
```

Comment:
Note sure if it's the right place to ask but

General question/issue, can the jsdoc of the `setBaggageItem` also specify that the string keys used need to respect the `http` header key format of `kebab-case`? 


We had a use case where we added a baggage item with a `camelCase` keyName only for it to give us the error which is self explanatory. But it would've been nice to know in the jsdoc. 
```
⨯ TypeError [ERR_INVALID_HTTP_TOKEN]: Header name must be a valid HTTP token ["ot-baggage-camelCase"]
```

Reply:
I don't think that is expected, as the header name provided in the error should actually be valid. Are you getting this error with HTTP 1 or HTTP 2?

Reply:
@rochdev thanks for the prompt response. our actual error code was it read in the error as `HTTP token ["ot-baggage-Symbol(next.rootSpanId)"]` we fixed it my having it be `next-span`